### PR TITLE
Closes #12 -- adds gh column to gh_decode output to reflect received …

### LINKED ISSUE #12: output of gh_decode should incorporate input
I've got starting point and stopping point geohashes that I aim to convert to lat/lon pairs for numeric analysis.

Wouldn't it be easy to (a) assign the input geohashes as row names (would by necessity be an option since duplicate row names are not allowed) or (b) create a `geohash` column by default which encapsulates the order in which inputs were processed?

Example:

```
library(data.table)
library(geohash)
DT = data.table(
  start = c("w21zux", "w21z6u", "w21ztq", "w21zkq", "w21z8s"),
  stop = c("w21ztq", "w21zkq", "w21z8s", "w21z7q", "w21xxs")
)

# all available geohashes to decode
all_geos = DT[ , unique(c(start, stop))]

geo_xy = gh_decode(all_geos)
# prepare for data.table join syntax
setDT(geo_xy)
geo_xy[ , geo := all_geos] # <- this step seems overkill

# first join
DT[geo_xy, c('start_lat', 'start_lon') := .(i.lat, i.lng),
    on = c(start = 'geo')]
# second join
DT[geo_xy, c('stop_lat', 'stop_lon') := .(i.lat, i.lng),
   on = c(stop = 'geo')]
DT[]
#     start   stop start_lat start_lon stop_lat stop_lon
# 1: w21zux w21ztq  1.403503  103.9142 1.354065 103.9471
# 2: w21z6u w21zkq  1.299133  103.8373 1.310120 103.9032
# 3: w21ztq w21z8s  1.354065  103.9471 1.343079 103.7384
# 4: w21zkq w21z7q  1.310120  103.9032 1.310120 103.8593
# 5: w21z8s w21xxs  1.343079  103.7384 1.343079 103.6945
```

In this case, it's just an inconvenience, but there are more dynamic use cases when it may be hard for the user to understand the order of output, and the convenience of this feature would be all the more.

IIUC would be as simple as changing [this command](https://github.com/Ironholds/geohash/blob/master/src/geohash.cpp#L124-L127):

```
  return DataFrame::create(_["gh"] = hashes,
                           _["lat"] = lats,
                           _["lng"] = lngs,
                           _["lat_error"] = lat_error,
                           _["lng_error"] = lng_error);
```

And adding tests; happy to file a PR if you agree it's useful/approve of the API

### PR DESCRIPTION
**Pull request by MichaelChirico**
…geohash names for convenience

happy to adjust the API as you see fit

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -1,8 +1,8 @@
 Package: geohash
 Type: Package
 Title: Tools for Geohash Creation and Manipulation
-Version: 0.3.0
-Date: 2017-08-02
+Version: 0.4.0
+Date: 2018-01-31
 Author: Oliver Keyes [aut, cre], Seth Wenchel [ctb], G. Nagel [cph]
```
Comment (Ironholds) [inline DESCRIPTION]: Add yerself as a contributor! you get credit for this :)

**Comment by MichaelChirico** [comment]
@Ironholds done; Travis also revealed another issue, which is that it appears `stringsAsFactors` is `TRUE` by default here, so `gh` comes back as a `factor`. I guess this is probably preferred (geohash is a `factor` variable if ever there was one), so leaving as is.

**Comment by Ironholds** [comment]
I'd rather it as a string, to be honest; it maintains package consistency and I'd rather users make factors as an explicit choice!

**Comment by MichaelChirico** [comment]
@Ironholds done

**Comment by Ironholds** [comment]
Perf; thanks!

**Pull request merged**